src/gt4py/eve/visitors.py

```diff
@@ -162,6 +165,15 @@ def generic_visit(self, node: concepts.RootNode, **kwargs: Any) -> Any:
                     if (new_child := self.visit(child, **kwargs)) is not NOTHING
                 },
             )
+            if self.PRESERVED_ANNEX_ATTRS:
+                new_annex = copy.deepcopy(new_node.annex)
+                for k in self.PRESERVED_ANNEX_ATTRS:
+                    try:
+                        assert not hasattr(new_annex, k)
+                        setattr(new_annex, k, getattr(node.annex, k))
+                    except AttributeError:
+                        pass
+                object.__setattr__(new_node, "__node_annex__", new_annex)
```

Comment:
I don't understand why is the new annex deep-copied before being filled. If I'm not missing something, the new node has been just initialized and thus it doesn't have data in the annex. 

Another issue I see here is that the try-except pattern used to copy the data is only fast in the case where most of the nodes contains all the annex attributes to be preserved, because the `except` branch is much slower than the `try` branch. I would suggest to use a different strategy to copy the annex data, something like: 
```suggestion
            if self.PRESERVED_ANNEX_ATTRS and (old_annex := getattr(node, "__node_annex__", None)):
                new_annex_dict = new_node.annex.__dict__
                for key in self.PRESERVED_ANNEX_ATTRS:
                    if value := getattr(old_annex, key, None):
                        new_annex_dict[key] = value
```

Reply:
Agreed. I was a little sloppy... I sticked to your suggestion, just added an assert that ensures a preserved attribute is not a computed one.